python/lib/communication/dmod/communication/maas_request/ngen/abstract_nextgen_request.py

```diff
@@ -103,43 +92,61 @@ def bmi_config_data_id(self) -> str:
         return self.request_body.bmi_config_data_id
 
     @property
-    def bmi_cfg_data_requirement(self) -> DataRequirement:
+    def catchments(self) -> Optional[List[str]]:
         """
-        A requirement object defining of the BMI configuration data needed to execute this request.
+        An optional list of catchment ids for those catchments in the request ngen execution.
+
+        No list implies "all" known catchments.
```

Comment:
Here, "all" means all catchment known in the hydrofabric.

Tangentially - and there's probably nothing we can do about it, but I am going to point it out in case we can, because as-is it is going to make the `git blame` history misleading - this wasn't really changed.  I.e., the `catchment` property definition, docstring, etc. existed like this lower in the file, and other property/function definitions above it were completely taken out.  The diff doesn't really represent that at all, though.